test/integration/kube_api_discovery_test.go

```diff
@@ -399,3 +402,109 @@ func TestGetAPIResourceList(t *testing.T) {
+	// assert which fields are set as AdditionalPrinterColumns.
+	// Ideally, every CRD should show some kind of identifying info, some kind of status, and Age.
+	expectedColumnsPerCRDVersion := map[string]map[string][]apiextensionsv1.CustomResourceColumnDefinition{
+		addSuffix("credentialissuers.config.concierge"): {
+			"v1alpha1": []apiextensionsv1.CustomResourceColumnDefinition{
+				{Name: "ProxyMode", Type: "string", JSONPath: ".spec.impersonationProxy.mode"},
+				// CredentialIssuers status is a list of strategies, each with its own status. Unfortunately,
+				// AdditionalPrinterColumns cannot show multiple results, e.g. a list of strategy types where
+				// the status is equal to Successful. See https://github.com/kubernetes/kubernetes/issues/67268.
+				// So we don't show any status as a AdditionalPrinterColumn at the moment.
+				{Name: "Age", Type: "date", JSONPath: ".metadata.creationTimestamp"},
+			},
+		},
+		addSuffix("webhookauthenticators.authentication.concierge"): {
+			"v1alpha1": []apiextensionsv1.CustomResourceColumnDefinition{
+				{Name: "Endpoint", Type: "string", JSONPath: ".spec.endpoint"},
+				// Note that WebhookAuthenticators have a status type, but no controller currently sets the status, so we don't show it.
+				{Name: "Age", Type: "date", JSONPath: ".metadata.creationTimestamp"},
+			},
+		},
+		addSuffix("jwtauthenticators.authentication.concierge"): {
+			"v1alpha1": []apiextensionsv1.CustomResourceColumnDefinition{
+				{Name: "Issuer", Type: "string", JSONPath: ".spec.issuer"},
+				{Name: "Audience", Type: "string", JSONPath: ".spec.audience"},
+				// Note that JWTAuthenticators have a status type, but no controller currently sets the status, so we don't show it.
+				{Name: "Age", Type: "date", JSONPath: ".metadata.creationTimestamp"},
+			},
+		},
+		addSuffix("activedirectoryidentityproviders.idp.supervisor"): {
+			"v1alpha1": []apiextensionsv1.CustomResourceColumnDefinition{
+				{Name: "Host", Type: "string", JSONPath: ".spec.host"},
+				{Name: "Status", Type: "string", JSONPath: ".status.phase"},
+				{Name: "Age", Type: "date", JSONPath: ".metadata.creationTimestamp"},
+			},
+		},
+		addSuffix("federationdomains.config.supervisor"): {
+			"v1alpha1": []apiextensionsv1.CustomResourceColumnDefinition{
+				{Name: "Issuer", Type: "string", JSONPath: ".spec.issuer"},
+				{Name: "Status", Type: "string", JSONPath: ".status.status"},
+				{Name: "Age", Type: "date", JSONPath: ".metadata.creationTimestamp"},
+			},
+		},
+		addSuffix("ldapidentityproviders.idp.supervisor"): {
+			"v1alpha1": []apiextensionsv1.CustomResourceColumnDefinition{
+				{Name: "Host", Type: "string", JSONPath: ".spec.host"},
+				{Name: "Status", Type: "string", JSONPath: ".status.phase"},
+				{Name: "Age", Type: "date", JSONPath: ".metadata.creationTimestamp"},
+			},
+		},
+		addSuffix("oidcidentityproviders.idp.supervisor"): {
```

Comment:
Maybe `allowPasswordGrant` would be valuable here as well?

Reply:
I was under the impression that the columns should only include the most important/identifying fields. For example. for Pods there are a zillion configuration options but they only show columns for Ready, Status, Restarts (which are all flavors of status) and Age. From that point of view `allowPasswordGrant` seems like a small detail as opposed to important/identifying.